apps/studio/components/interfaces/HomeNew/GettingStarted/GettingStartedSection.tsx

```diff
@@ -0,0 +1,563 @@
+import {
+  AiIconAnimation,
+  Button,
+  Card,
+  CardContent,
+  CodeBlock,
+  ToggleGroup,
+  ToggleGroupItem,
+} from 'ui'
+import { BASE_PATH } from 'lib/constants'
+
+export type GettingStartedAction = {
+  label: string
+  href?: string
+  onClick?: () => void
+  variant?: React.ComponentProps<typeof Button>['type']
+  icon?: React.ReactNode
+  component?: React.ReactNode
+}
+
+export type GettingStartedStep = {
+  key: string
+  status: 'complete' | 'incomplete'
+  icon?: React.ReactNode
+  title: string
+  description: string
+  image?: React.ReactNode
+  actions: GettingStartedAction[]
+}
+
+type GettingStartedState = 'empty' | 'code' | 'no-code' | 'hidden'
+
+export function GettingStartedSection({
+  value,
+  onChange,
+}: {
+  value: GettingStartedState
+  onChange: (v: GettingStartedState) => void
+}) {
+  const { data: project } = useSelectedProjectQuery()
+  const { ref } = useParams()
+  const aiSnap = useAiAssistantStateSnapshot()
+  const router = useRouter()
+
+  // Local state for framework selector preview
+  const [selectedFramework, setSelectedFramework] = useState<string>(FRAMEWORKS[0]?.key ?? 'nextjs')
+  const workflow: 'no-code' | 'code' | null = value === 'code' || value === 'no-code' ? value : null
+
+  const { data: tablesData } = useTablesQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+    schema: 'public',
+  })
+
+  const tablesCount = Math.max(0, tablesData?.length ?? 0)
+  const { data: branchesData } = useBranchesQuery({
+    projectRef: project?.parent_project_ref ?? project?.ref,
+  })
+  const isDefaultProject = project?.parent_project_ref === undefined
+  const hasNonDefaultBranch =
+    (branchesData ?? []).some((b) => !b.is_default) || isDefaultProject === false
+
+  // Helpers
+  const openAiChat = useCallback(
+    (name: string, initialInput: string) => aiSnap.newChat({ name, open: true, initialInput }),
+    [aiSnap]
+  )
+
+  const openConnect = useCallback(() => {
+    router.push(
+      {
+        pathname: router.pathname,
+        query: {
+          ...router.query,
+          showConnect: true,
+          connectTab: 'frameworks',
+          framework: selectedFramework,
+        },
+      },
+      undefined,
+      { shallow: true }
+    )
+  }, [router, selectedFramework])
+
+  const connectActions: GettingStartedAction[] = useMemo(
+    () => [
+      {
+        label: 'Framework selector',
+        component: (
+          <FrameworkSelector
+            value={selectedFramework}
+            onChange={setSelectedFramework}
+            items={FRAMEWORKS}
+          />
+        ),
+      },
+      {
+        label: 'Connect',
+        variant: 'default',
+        onClick: openConnect,
+      },
+    ],
+    [openConnect, selectedFramework]
+  )
+
+  const codeSteps: GettingStartedStep[] = useMemo(
+    () => [
+      {
+        key: 'install-cli',
+        status: 'incomplete',
```

Comment:
Should there be a way to manually mark these steps (that can't be auto-checked) as complete? I personally would find it mildly infuriating to have them sitting around as incomplete perpetually 😄